### PR TITLE
Adkernel: Add static endpoint

### PR DESCRIPTION
**Pull request by marki1an**
https://github.com/prebid/prebid-server/pull/2169

### CODE REVIEW

File: src/test/java/org/prebid/server/bidder/adkernel/AdkernelBidderTest.java [hunk 1]
```diff
@@ -42,7 +42,7 @@
 
 public class AdkernelBidderTest extends VertxTest {
 
-    private static final String ENDPOINT_URL = "http://%s/hb?zone=%s";
+    private static final String ENDPOINT_URL = "https://pbs.adksrv.com/hb?zone=%s";
```
Comment (And1sS) [inline src/test/java/org/prebid/server/bidder/adkernel/AdkernelBidderTest.java:45]: We don't use real endpoints in tests, please, replace with something like `"https://test.com?zone=%s"`

File: src/test/java/org/prebid/server/bidder/adkernel/AdkernelBidderTest.java [hunk 2]
```diff
@@ -70,7 +70,8 @@ public void makeHttpRequestsShouldReturnErrorWhenImpHasNoBannerOrVideo() {
         // then
         assertThat(result.getValue()).isEmpty();
         assertThat(result.getErrors())
-                .containsExactly(BidderError.badInput("Invalid imp id=123. Expected imp.banner or imp.video"));
+                .containsExactly(BidderError
+                        .badInput("Invalid imp id=123. Expected imp.banner / imp.video / imp.native"));
```
Comment (And1sS) [inline src/test/java/org/prebid/server/bidder/adkernel/AdkernelBidderTest.java:74]: Please, place BidderError.badInput on the same line.

File: src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java [hunk 3]
```diff
@@ -79,9 +78,9 @@ private void processImp(Imp imp, Map<ExtImpAdkernel, List<Imp>> pubToImps) {
     }
 
     private static void validateImp(Imp imp) {
-        if (imp.getBanner() == null && imp.getVideo() == null) {
+        if (imp.getBanner() == null && imp.getVideo() == null && imp.getXNative() == null) {
             throw new PreBidException(String.format(
-                    "Invalid imp id=%s. Expected imp.banner or imp.video", imp.getId()));
+                    "Invalid imp id=%s. Expected imp.banner / imp.video / imp.native", imp.getId()));
```
Comment (And1sS) [inline src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java:83]: Can be simplified to `"Invalid imp id=" + imp.getId() + ". Expected imp.banner / imp.video / imp.native"`.

**Review by And1sS — CHANGES_REQUESTED** [review]

**Review by And1sS — APPROVED** [review]

File: src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java [hunk 4]
```diff
@@ -112,10 +108,17 @@ private static void dispatchImpression(Imp imp, ExtImpAdkernel extImpAdkernel,
...
+
+        if (imp.getBanner() != null) {
+            return impBuilder.video(null).xNative(null).build();
+        } else if (imp.getVideo() != null) {
+            return impBuilder.banner(null).xNative(null).build();
+        } else if (imp.getXNative() != null) {
+            return impBuilder.banner(null).video(null).build();
+        }
+
+        throw new PreBidException("Invalid impression");
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java:121]: Not possible case due to validateImp method, last with native can be treated as default

File: src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java [hunk 5]
```diff
@@ -112,10 +108,17 @@ private static void dispatchImpression(Imp imp, ExtImpAdkernel extImpAdkernel,
...
-                .xNative(null);
-        return imp.getBanner() != null ? impBuilder.video(null).build() : impBuilder.build();
+        final Imp.ImpBuilder impBuilder = imp.toBuilder().ext(null).audio(null);
+
+        if (imp.getBanner() != null) {
+            return impBuilder.video(null).xNative(null).build();
+        } else if (imp.getVideo() != null) {
+            return impBuilder.banner(null).xNative(null).build();
+        } else if (imp.getXNative() != null) {
+            return impBuilder.banner(null).video(null).build();
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java:118]: Due to if conditions higher banner and video already null, pls fix for 116 and 118 line

File: src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java [hunk 6]
```diff
@@ -79,9 +78,9 @@ private void processImp(Imp imp, Map<ExtImpAdkernel, List<Imp>> pubToImps) {
     }
 
     private static void validateImp(Imp imp) {
-        if (imp.getBanner() == null && imp.getVideo() == null) {
-            throw new PreBidException(String.format(
-                    "Invalid imp id=%s. Expected imp.banner or imp.video", imp.getId()));
+        if (imp.getBanner() == null && imp.getVideo() == null && imp.getXNative() == null) {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java:81]: Should we update `media-types` in adkernel.yaml?

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

File: src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java (hunk 6 above)
Comment (marki1an) [reply src/main/java/org/prebid/server/bidder/adkernel/AdkernelBidder.java:81]: at the current time not

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**